react_juce/core/EcmascriptEngine_Hermes.cpp

```diff
@@ -0,0 +1,522 @@
+
+                const jsi::Value *jsiArgsPtr = jsiArgs.data();
+                return jsiValueToVar(fPtr->fn.call(rt, jsiArgsPtr, jsiArgs.size()), rt);
+            };
+        }
+
+        juce::var jsiObjectToVarObject(jsi::HostObject &v, jsi::Runtime &runtime)
+        {
+            juce::DynamicObject::Ptr varObj = new juce::DynamicObject();
+
+            auto props = v.getPropertyNames(runtime);
+            for (auto &propID : props)
+            {
+                jsi::Value property = v.get(runtime, propID);
+                varObj->setProperty(juce::String(propID.utf8(runtime)), jsiValueToVar(property, runtime));
+            }
+
+            return varObj.get();
+        }
+
+        juce::var jsiValueToVar(const jsi::Value &v, jsi::Runtime &runtime)
+        {
+            if (v.isBool())
+                return v.getBool();
+
+            if (v.isNumber())
+                return v.getNumber();
+
+            if (v.isString())
+                return juce::String(v.getString(runtime).utf8(runtime));
+
+            if (v.isUndefined())
+                return juce::var::undefined();
+
+            if (v.isNull())
+                return juce::var();
+
+            if (v.isObject())
+            {
+                auto obj = v.getObject(runtime);
+
+                if (obj.isArray(runtime))
+                    return jsiArrayToVarArray(obj.getArray(runtime), runtime);
+
+                if (obj.isFunction(runtime))
+                    return jsiMethodToVarMethod(obj.asFunction(runtime), runtime);
+
+                if (obj.isHostObject(runtime))
+                {
+                    auto hostObjectPtr = obj.getHostObject(runtime);
+                    return jsiObjectToVarObject(*hostObjectPtr, runtime);
+                }
+
+                if (obj.isArrayBuffer(runtime))
+                {
+                    jsi::ArrayBuffer arrayBuffer = obj.getArrayBuffer(runtime);
+                    juce::MemoryBlock memBlock(static_cast<const void*>(arrayBuffer.data(runtime)),
+                                                                        arrayBuffer.size(runtime));
+
+                    return juce::var(memBlock);
+                }
+            }
+
+            return {};
+        }
+
+        //==============================================================================
+        jsi::Object varObjectToJSIObject(juce::DynamicObject *v, jsi::Runtime &runtime)
+        {
+            jassert(v);
+
+            jsi::Object jsiObj(runtime);
+            for (auto &prop : v->getProperties())
+            {
+                jsiObj.setProperty(runtime, prop.name.getCharPointer(), varToJSIValue(prop.value, runtime));
+            }
+
+            return jsiObj;
+        }
+
+        jsi::Function varMethodToJSIMethod(juce::var::NativeFunction &v, jsi::Runtime &runtime)
+        {
+            return jsi::Function::createFromHostFunction(
+                runtime,
+                jsi::PropNameID::forAscii(runtime, ""),
```

Comment:
What is this?

Reply:
Tis this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name

Reply:
I think it's OK to leave empty otherwise the implementation needs to be a bit more painful ... ?

Reply:
In React Native I believe you'll see something like "[Native Method]" for the name here

Reply:
Right yeah. Do we want to modify to match?

Reply:
@nick-thompson, just in case this one gets lost in the weeds. Did you want to update the function name property here to `ReactJuceNative` or something?

Reply:
I think so, only because it will show up in stack traces and console logs if the user needs

Reply:
Yeah cool. Will Add. "[ReactJuce Native Method]".